orttraining/orttraining/training_api/include/onnxruntime_training_c_api.h

```diff
@@ -221,6 +221,58 @@ struct OrtTrainingApi {
   */
   ORT_API2_STATUS(SchedulerStep, _Inout_ OrtTrainingSession* sess);
 
+  /** \brief Retrieves the size of all the parameters.
+  *
+  * Calculates the size of all the parameters for the training session.
+  * When 'trainable_only' is true, the size is calculated for trainable params only.
+  *
+  * \param[in] sess The training session.
+  * \param[in] trainable_only Whether to skip non-trainable parameters
+  *
+  * \snippet{doc} snippets.dox OrtStatus Return Value
+  *
+  */
+  ORT_API2_STATUS(GetParametersSize, _Inout_ OrtTrainingSession* sess,
+                  _Out_ size_t* out, bool trainable_only);
+
+  /** \brief Copy parameters onto contiguous buffer held by parameters_buffer
+  *
+  * The parameters_buffer has to be of the size given by GetParametersSize api call,
+  * with matching setting for 'trainable_only'. All the target parameters must be of the same
+  * datatype. The OrtValue must be pre-allocated onto
+  * the desired device. This is a complementary function to 'CopyBufferToParameters'.
+  * Parameter ordering is preserved.
+  * User is responsible for allocating/freeing the 'parameters_buffer'.
+  *
+  * \param[in] sess The training session.
+  * \param[in] trainable_only Whether to skip non-trainable parameters
+  * \param[out] parameters_buffer The pre-allocated OrtValue buffer to copy onto.
+  *
+  * \snippet{doc} snippets.dox OrtStatus Return Value
+  *
+  */
+  ORT_API2_STATUS(CopyParametersToBuffer, _Inout_ OrtTrainingSession* sess,
+                    _Inout_ OrtValue& parameters_buffer, bool trainable_only);
+
+  /** \brief Copy parameter values from contiguous buffer held by parameters_buffer onto parameters
+  *
+  * The parameters_buffer has to be of the size given by GetParametersSize api call,
+  * with matching setting for 'trainable_only'. All the target parameters must be of the same
+  * datatype. This is a complementary function to 'CopyBufferToParameters'
+  * and can be used to load updated buffer values onto the parameters. 
+  * Parameter ordering is preserved.
+  * User is responsible for allocating/freeing the 'parameters_buffer'.
+  *
+  * \param[in] sess The training session.
+  * \param[in] trainable_only Whether to skip non-trainable parameters
+  * \param[out] parameters_buffer The pre-allocated OrtValue buffer to copy from.
+  *
+  * \snippet{doc} snippets.dox OrtStatus Return Value
+  *
+  */
+  ORT_API2_STATUS(CopyBufferToParameters, _Inout_ OrtTrainingSession* sess,
+                    _Inout_ OrtValue& parameters_buffer, bool trainable_only);
```

Comment:
`OrtValue` here might need to be a pointer considering that C doesnt allow passing by reference.